Furball.Engine/Engine/Graphics/Drawables/Primitives/LinePrimitiveDrawable.cs

```diff
@@ -1,38 +1,26 @@
+using System.Numerics;
 using Furball.Engine.Engine.Graphics.Drawables.Managers;
-using Microsoft.Xna.Framework;
-using Xssp.MonoGame.Primitives2D;
+using Furball.Vixie.Graphics;
 
 namespace Furball.Engine.Engine.Graphics.Drawables.Primitives {
     /// <summary>
     /// Simple Line Drawable
     /// </summary>
     public class LinePrimitiveDrawable : ManagedDrawable {
-        /// <summary>
-        ///     The length of the line in pixels
-        /// </summary>
-        public float Length;
-        /// <summary>
-        ///     The thickness of the line
-        /// </summary>
-        public float Thickness = 1f;
-        /// <summary>
-        ///     The angle of the line in radians
-        /// </summary>
-        public float Angle;
-        /// <summary>
-        /// Creates a Line
-        /// </summary>
-        /// <param name="position">Where to Draw</param>
-        /// <param name="length">The length of the line</param>
-        /// <param name="angle">The angle of the line in radians</param>
-        public LinePrimitiveDrawable(Vector2 position, float length, float angle) {
-            this.Position = position;
-            this.Length   = length;
-            this.Angle    = angle;
+        public Vector2 EndPosition;
+        public float   Thickness;
+        public bool    RelativePosition;
+
+        public LinePrimitiveDrawable(Vector2 start, Vector2 end, Color color, float thickness = 1f, bool relative = true) {
+            this.Position         = start;
+            this.EndPosition      = end;
+            this.ColorOverride    = color;
+            this.Thickness        = thickness;
+            this.RelativePosition = relative;
```

Comment:
While the `RelativePosition` variable solves one issue where you couldnt move a line at all without p2 being stuck, it still doesnt solve all the issues, the main one being, you are unable to independantly move the 2 points with Tweens, so for example i want p1 to go from 0,0 to 100,100 and p2 to go from 100,200 to 200,500. i couldnt do that without a lot of very complicated math, there needs to be a way for drawables to add their own tweenable values

Reply:
i have no idea where to even begin on that, and i dont think this should be specific to this pr

Reply:
> i have no idea where to even begin on that, and i dont think this should be specific to this pr given this isnt something vixie related

I guess, but its someone that should have an open issue tracking it once this PR is merged